src/main/java/org/opensearch/security/dlic/rest/api/AccountApiAction.java

```diff
@@ -196,30 +239,40 @@ protected void handlePut(RestChannel channel, final RestRequest request, final C
         }
 
         final SecurityJsonNode securityJsonNode = new SecurityJsonNode(content);
-        final String currentPassword = content.get("current_password").asText();
         final Hashed internalUserEntry = (Hashed) internalUser.getCEntry(username);
-        final String currentHash = internalUserEntry.getHash();
-
-        if (currentHash == null || !OpenBSDBCrypt.checkPassword(currentHash, currentPassword.toCharArray())) {
-            badRequestResponse(channel, "Could not validate your current password.");
-            return;
-        }
-
-        // if password is set, it takes precedence over hash
-        final String password = securityJsonNode.get("password").asString();
-        final String hash;
-        if (Strings.isNullOrEmpty(password)) {
-            hash = securityJsonNode.get("hash").asString();
-        } else {
-            hash = hash(password.toCharArray());
-        }
-        if (Strings.isNullOrEmpty(hash)) {
-            badRequestResponse(channel, "Both provided password and hash cannot be null/empty.");
-            return;
+        final SecurityDynamicConfiguration<?> configuration = load(getConfigName(), false);
+        if (request.path().endsWith(SAVED_TENANT)){
+            if (configuration.exists(user.getName())){
+                InternalUserV7 iu = (InternalUserV7) internalUser.getCEntry(username);
```

Comment:
Same as before (InternalUserV6 ?)

Reply:
Should we validate if user has access to the tenant which user wishes to set?

Reply:
I believe we do not need to (those bad cases should be handled by the front end), but I'll work on that in case that assumption about the front end does not hold true.

Reply:
As this is an API and can be triggered directly, we should validate that. I believe there is already a functionality which checks that. `hasTenantPermission`, please check if you can reuse that